sentinel/lib/schedules.js

```diff
@@ -120,13 +120,35 @@ module.exports = {
                 if (due < now || !message) {
                     return;
                 }
-                messages.scheduleMessage(doc, {
-                    due: due.toISOString(),
-                    message: message,
-                    group: msg.group,
-                    type: schedule.name,
-                    translation_key: schedule.translation_key
-                }, phone, registrations, patient);
+
+                try {
+                    const task = {
+                        due: due.toISOString(),
+                        group: msg.group,
+                        type: schedule.name,
+                        translation_key: schedule.translation_key,
+                        message_key: msg.translation_key,
+                        recipient: msg.recipient
+                    };
+                    if (!msg.translation_key) {
+                        // no translation_key so generate now
```

Comment:
Does this mean that deferred translation doesn't happen if a translation key was not configured?  This would be surprising.

Reply:
Yeah. We want everyone to move to using `translation_key` anyway so I think it's ok. The problem was with supporting the `messages` array we'd have to inline the whole array into the doc so we could generate the message just in time. I'll make sure I document this requirement somewhere in the release notes.